lib/Sema/CSApply.cpp

```diff
@@ -3432,6 +3432,11 @@ namespace {
               auto &tc = cs.getTypeChecker();
               tc.diagnose(cast->getLoc(), diag::conditional_downcast_foreign,
                           destValueType);
+              ConcreteDeclRef refDecl = sub->getReferencedDecl();
+              if (refDecl) {
```

Comment:
Almost. You want to check `getDecl()` here.

Reply:
@CodaFi Thank you for guiding me till here. As I am still new to Cpp development I would like to clarify following things:
* Do we need to check the result of `getBaseName` to be not null as well?
* I found some code at [MiscDiagnostics](https://github.com/apple/swift/blob/master/lib/Sema/MiscDiagnostics.cpp#L1722) which is using a full chain inside diagnose method. Isn't it not safe too?

Reply:
- `nullptr` is not an inhabitant of `DeclBaseName` 
- It is, but only if truly malformed AST manages to slip through all of Sema into CSApply, so it's morally correct.  My concern here is that you may not necessarily be diagnosing a declaration referenced by the cast target - you could very well have a pure expression in this position - so you will get `ConcreteDeclRef()` which has a null referenced declaration.

Reply:
Ah, wait, there's an `operator bool()` here.

:shipit: